// lightgallery/angular/15/esm2022/lib/lightgallery-angular.module.mjs
import { NgModule } from '@angular/core';
import { LightgalleryComponent } from './lightgallery-angular.component';
import * as i0 from "@angular/core";
class LightgalleryModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: LightgalleryModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: LightgalleryModule, declarations: [LightgalleryComponent], exports: [LightgalleryComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: LightgalleryModule }); }
}
export { LightgalleryModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: LightgalleryModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LightgalleryComponent],
                    imports: [],
                    exports: [LightgalleryComponent],
                }]
        }] });
