// html-validate/dist/es/browser.js
export { c as compatibilityCheck } from './core-browser.js';
export { A as Attribute, C as Config, a as ConfigError, b as ConfigLoader, D as DOMNode, h as DOMTokenList, j as DOMTree, k as DynamicValue, I as EventHandler, l as HtmlElement, H as HtmlValidate, M as MetaCopyableProperty, q as MetaTable, n as NestedError, N as NodeClosed, m as NodeType, P as Parser, B as Reporter, R as ResolvedConfig, t as Rule, o as SchemaValidationError, S as Severity, f as StaticConfigLoader, w as TextClassification, r as TextContent, T as TextNode, U as UserError, V as Validator, W as WrappedError, u as ariaNaming, x as classifyNodeText, p as configPresets, c as defineConfig, E as definePlugin, y as keywordPatternMatcher, F as ruleExists, z as sliceLocation, s as staticResolver, v as version, G as walk } from './core.js';
export { d as defineMetadata, m as metadataHelper } from './meta-helper.js';
import 'ajv';
import './elements.js';
import '@sidvind/better-ajv-errors';
import './utils/natural-join.js';
import 'fs';
import 'kleur';
import '@html-validate/stylish';
import 'semver';
